lib/Conversion/XTenNNToTorch.cpp

```diff
@@ -37,20 +37,25 @@ using namespace mlir::torch;
 
 namespace {
 
-Value toTorchTensorTypeCast(PatternRewriter &rewriter, Value input) {
+Type toTorchTensorTypeCast(PatternRewriter &rewriter, ShapedType ty) {
+  auto elementType = ty.getElementType();
 
-  auto tensorTy = dyn_cast<ShapedType>(input.getType());
-  auto sizes = tensorTy.getShape();
-  auto tensorEltTy = tensorTy.getElementType();
-  if (tensorEltTy.isSignlessInteger()) {
-    tensorEltTy = rewriter.getIntegerType(tensorTy.getElementTypeBitWidth(), true);
+  auto intElementType = dyn_cast<IntegerType>(ty.getElementType());
+  if (intElementType && intElementType.isSignlessInteger() && intElementType.getWidth() != 1) {
+      elementType = rewriter.getIntegerType(elementType.getIntOrFloatBitWidth(),
+        /*isSigned=*/true);
   }
 
+  return Torch::ValueTensorType::get(
+      ty.getContext(), ty.getShape(), elementType);
+}
+
+Value toTorchTensorTypeCast(PatternRewriter &rewriter, Value input) {
+  auto tensorTy = dyn_cast<ShapedType>(input.getType());
```

Comment:
Should we have error handling here if the cast is not successful? The code afterwards accesses it unconditionally, so this should either be a cast or bail out on if the dyn_cast does not succeed.

(just noticed in `matchAndRewrite` there is a hard cast)

Reply:
This should be a `cast` indeed. Like in the first commit :)